From e6b6b6e99a4bd4a9ee9bda28068df10aff78859d Mon Sep 17 00:00:00 2001
Message-Id: <e6b6b6e99a4bd4a9ee9bda28068df10aff78859d.1476720588.git.roliveir@synopsys.com>
From: Ramiro Oliveira <roliveir@synopsys.com>
Date: Mon, 17 Oct 2016 17:08:56 +0100
Subject: [PATCH] Add a sysfs file.

Signed-off-by: Ramiro Oliveira <roliveir@synopsys.com>
---
 drivers/net/ethernet/intel/e1000/e1000_main.c | 36 +++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/drivers/net/ethernet/intel/e1000/e1000_main.c b/drivers/net/ethernet/intel/e1000/e1000_main.c
index f42129d..35e2a63 100644
--- a/drivers/net/ethernet/intel/e1000/e1000_main.c
+++ b/drivers/net/ethernet/intel/e1000/e1000_main.c
@@ -33,6 +33,8 @@
 #include <linux/bitops.h>
 #include <linux/if_vlan.h>
 
+#define EUDYPTULA_ID "2a86d024c0e5"
+
 char e1000_driver_name[] = "e1000";
 static char e1000_driver_string[] = "Intel(R) PRO/1000 Network Driver";
 #define DRV_VERSION "7.3.21-k8-NAPI"
@@ -229,6 +231,39 @@ static int debug = -1;
 module_param(debug, int, 0);
 MODULE_PARM_DESC(debug, "Debug level (0=none,...,16=all)");
 
+static ssize_t eudyptula_id_show(struct device *dev,
+				 struct device_attribute *attr, char *buf)
+{
+	return sprintf(buf, "%s\n", EUDYPTULA_ID);
+}
+
+static ssize_t eudyptula_id_store(struct device *dev,
+				  struct device_attribute *attr, 
+				  const char *buf, size_t count)
+{
+	int ret;
+
+	ret = strncmp(buf, EUDYPTULA_ID, strlen(EUDYPTULA_ID));
+
+	if (ret != 0)
+		return -EINVAL;
+
+	return count;
+}
+
+static DEVICE_ATTR_RW(eudyptula_id);
+
+static struct attribute *attrs[] = {
+	&dev_attr_eudyptula_id.attr,
+	NULL,	/* need to NULL terminate the list of attributes */
+};
+
+static struct attribute_group eudyptula_attr_group = {
+	.name = "eudyptula",
+	.attrs = attrs,
+};
+
+
 /**
  * e1000_get_hw_dev - return device
  * used by hardware layer to print debugging information
@@ -1212,6 +1247,7 @@ static int e1000_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 	e1000_reset(adapter);
 
 	strcpy(netdev->name, "eth%d");
+	netdev->sysfs_groups[0] = &eudyptula_attr_group;
 	err = register_netdev(netdev);
 	if (err)
 		goto err_register;
-- 
2.9.3

