diff --git a/arch/x86/entry/syscalls/syscall_32.tbl b/arch/x86/entry/syscalls/syscall_32.tbl
index 2b36185..d2f7561 100644
--- a/arch/x86/entry/syscalls/syscall_32.tbl
+++ b/arch/x86/entry/syscalls/syscall_32.tbl
@@ -389,3 +389,4 @@
 380	i386	pkey_mprotect		sys_pkey_mprotect
 381	i386	pkey_alloc		sys_pkey_alloc
 382	i386	pkey_free		sys_pkey_free
+383	i386	eudyptula		sys_eudyptula
diff --git a/arch/x86/entry/syscalls/syscall_64.tbl b/arch/x86/entry/syscalls/syscall_64.tbl
index e93ef0b..b86fe05 100644
--- a/arch/x86/entry/syscalls/syscall_64.tbl
+++ b/arch/x86/entry/syscalls/syscall_64.tbl
@@ -338,6 +338,7 @@
 329	common	pkey_mprotect		sys_pkey_mprotect
 330	common	pkey_alloc		sys_pkey_alloc
 331	common	pkey_free		sys_pkey_free
+332	common	eudyptula		sys_eudyptula
 
 #
 # x32-specific system call numbers start at 512 to avoid cache impact
diff --git a/include/linux/syscalls.h b/include/linux/syscalls.h
index 91a740f..2a4d37d 100644
--- a/include/linux/syscalls.h
+++ b/include/linux/syscalls.h
@@ -902,5 +902,6 @@ asmlinkage long sys_pkey_mprotect(unsigned long start, size_t len,
 				  unsigned long prot, int pkey);
 asmlinkage long sys_pkey_alloc(unsigned long flags, unsigned long init_val);
 asmlinkage long sys_pkey_free(int pkey);
+asmlinkage long sys_eudyptula(int high_id, int low_id);
 
 #endif
diff --git a/init/Kconfig b/init/Kconfig
index 34407f1..2d9ba0b 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -2127,4 +2127,11 @@ config ASN1
 	  inform it as to what tags are to be expected in a stream and what
 	  functions to call on what tags.
 
+config EUDYPTULA_SYSCALL
+	bool "Add Eudyptula Challenge Syscall"
+	help
+	  Add the eudyptula challenge syscall to the kernel. This syscall will 
+	  look at two values, mush them together, and then compare with the user
+	  eudyptula ID.	
+
 source "kernel/Kconfig.locks"
diff --git a/init/Makefile b/init/Makefile
index c4fb455..466799e 100644
--- a/init/Makefile
+++ b/init/Makefile
@@ -16,6 +16,8 @@ ifneq ($(CONFIG_ARCH_INIT_TASK),y)
 obj-y                          += init_task.o
 endif
 
+obj-$(CONFIG_EUDYPTULA_SYSCALL) += eudyptula.o
+
 mounts-y			:= do_mounts.o
 mounts-$(CONFIG_BLK_DEV_RAM)	+= do_mounts_rd.o
 mounts-$(CONFIG_BLK_DEV_INITRD)	+= do_mounts_initrd.o
diff --git a/kernel/sys_ni.c b/kernel/sys_ni.c
index 635482e..e8087fb 100644
--- a/kernel/sys_ni.c
+++ b/kernel/sys_ni.c
@@ -255,3 +255,6 @@ cond_syscall(sys_membarrier);
 cond_syscall(sys_pkey_mprotect);
 cond_syscall(sys_pkey_alloc);
 cond_syscall(sys_pkey_free);
+
+/* Eudyptula syscall */
+cond_syscall(sys_eudyptula);
